src/main.rs

```diff
@@ -221,6 +246,76 @@ fn dump_lagrange(opts: DumpLagrangeOpts) {
     println!("srs_lagrange_form saved to {}", opts.srs_lagrange_form);
 }
 
+#[cfg(feature = "server")]
+fn prove_server(opts: ServerOpts) {
+    let circuit_file = resolve_circuit_file(opts.circuit);
+    println!("Loading circuit from {}...", circuit_file);
+    let circuit_base = CircomCircuit {
+        r1cs: reader::load_r1cs(&circuit_file),
+        witness: None,
+        wire_mapping: None,
+        aux_offset: plonk::AUX_OFFSET,
+    };
+
+    let srs_monomial_form = opts.srs_monomial_form;
+    let srs_lagrange_form = opts.srs_lagrange_form;
+
+    let builder = move || -> server::ProveCore {
+        //here we do setup, which should take some time
+        let setup = plonk::SetupForProver::prepare_setup_for_prover(
+            circuit_base.clone(),
+            reader::load_key_monomial_form(&srs_monomial_form),
+            reader::maybe_load_key_lagrange_form(srs_lagrange_form),
+        )
+        .expect("prepare err");
+
+        Box::new(move |witness: Vec<u8>, validate_only: bool| -> server::CoreResult {
+            let perf_t = std::time::SystemTime::now();
+
+            let mut circut = circuit_base.clone();
+            let witness_ret = reader::load_witness_from_array::<Bn256>(witness);
+            if let Ok(witness) = witness_ret {
+                circut.witness = Some(witness);
+            } else {
+                return server::CoreResult::any_prove_error(witness_ret, validate_only);
+            }
+            let proof_ret = setup.prove(circut);
```

Comment:
Here it actually proves  every time, and proving is very time-consuming.

we'd better add a conditional branch for https://github.com/Fluidex/plonkit/blob/006e688997eba2bd3687505999a213cf1eb93336/src/main.rs#L260-L270

Reply:
Ignore me. I find that only **prove** is requsted from the spec.

Reply:
Ignore me. I find that only **prove** is requsted from the spec.